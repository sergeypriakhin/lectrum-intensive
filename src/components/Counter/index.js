import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Styles from './styles.scss';


export default class Counter extends Component {

    static propTypes = {
        postCount: PropTypes.number.isRequired
    };

    render () {
        const { postCount } = this.props;

        return (
            <section className = { Styles.counter }>
                <span>
                    Post count: { postCount }
                </span>
            </section>
        );
    }
}
