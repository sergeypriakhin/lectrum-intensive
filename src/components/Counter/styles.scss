@import '../../theme/palette';

.counter {
  color: $paletteColor7;
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  span {
    position: relative;
    &::after {
      content: '';
      position: absolute;
      top: -6px;
      left: -35px;
      width: 30px;
      height: 30px;
      background-image: url(../../theme/assets/post.png);
      background-repeat: no-repeat;
      background-size: 30px;
    }
  }

}