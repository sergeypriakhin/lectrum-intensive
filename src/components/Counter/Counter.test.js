import React from 'react';
import Counter from './';
import dom from 'react-test-renderer';

const renderTree = dom.create(
    <Counter postCount = { 3 } />
).toJSON();

test('Counter component should correspond to its snapshot counterpart', () => {
    expect(renderTree).toMatchSnapshot(true);
});
