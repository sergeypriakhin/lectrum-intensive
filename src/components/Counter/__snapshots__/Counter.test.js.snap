// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`true 1`] = `
<section
  className={undefined}
>
  <span>
    Post count: 
    3
  </span>
</section>
`;
